### ISSUE TITLE
Support for NotBeAssignableTo

### ISSUE THREAD

**Issue by jnyrup** [fluentassertions/fluentassertions#594]
There does not seem to be support for the negation of `BeAssignableTo<T>`.

I propose addition of `NotBeAssignableTo<T>()`.

If the proposal is welcomed, I'll be happy to submit a PR once 5.0 is merged into master (unless somebody beats me to it).

**Comment by dennisdoomen** [comment]
Well, we need to merge *master* to *develop* anyway, so go ahead with that PR on *master.

**Comment by dennisdoomen** [comment]
Scratch that. If you can wait a couple of weeks and are fine with a temporary 5.0 version, you can target the PR at *develop*

**Comment by jnyrup** [comment]
I already have some implementations and tests ready for my own use, so I'm in no hurry.
I would just like to avoid having to rebase PRs after a large merge. 
So whatever causes me the fewest conflicts I'm happy with.

By "wait a couple of weeks " do then mean:
* "wait a couple of weeks before branching out", 
* "branch out now and wait a couple of weeks extra before it will be released",
* or something else?

**Comment by dennisdoomen** [comment]
"branch out now and wait a couple of weeks extra before it will be released",

**Comment by jnyrup** [comment]
closed in #601
